### PR TITLE
[internal] Abstracted isResponseEmpty conditional in a separate method

### PR DESCRIPTION
**Pull request by Zizzamia**
### Motivation
I want to help with some Architecture clean up to help make RestHooks's codebase easier to read and to contribute from a large audience.

Some nit:
 - logic abstraction
 - fix some spacing
 - fix some style comments

I will help with more general nits before I get into the real problems, let me know @ntucker if you don't mind those changes.

Thank you for this great library.

### CODE REVIEW

**Comment by Zizzamia** [comment]
Oh and let me know if you prefer different naming or coding changes.

File: src/resource/__tests__/resource.ts [hunk 1]
```diff
@@ -5,7 +5,9 @@ import {
   UserResource,
   UrlArticleResource,
 } from '../../__tests__/common';
-import { Resource, normalize, SimpleResource } from '..';
+import { normalize } from '../normal';
+import Resource, { isResponseEmpty } from '../Resource';
```
Comment (ntucker) [inline src/resource/__tests__/resource.ts]: are you using isResponseEmpty anywhere in this file?

Comment (Zizzamia) [reply src/resource/__tests__/resource.ts]: Oh no, my bad!
I was going to add more tests, but I decided to keep this PR small for now and do more tests in future PRs.
Let me remove `isResponseEmpty` for now.

File: src/resource/Resource.ts [hunk 2]
```diff
@@ -3,7 +3,20 @@ import { Method } from '~/types';
 
 import SimpleResource from './SimpleResource';
 
-/** Represents an entity to be retrieved from a server. Typically 1:1 with a url endpoint. */
+const ResourceError = `JSON expected but not returned from API`;
+
+export const isResponseEmpty = (res: request.Response):boolean => {
+  const resIncludesJSON = res.type.includes('json');
```
Comment (Zizzamia) [inline src/resource/Resource.ts]: Move this to line 13 :D

File: src/resource/Resource.ts [hunk 3]
```diff
@@ -3,7 +3,20 @@ import { Method } from '~/types';
 
 import SimpleResource from './SimpleResource';
 
-/** Represents an entity to be retrieved from a server. Typically 1:1 with a url endpoint. */
+const ResourceError = `JSON expected but not returned from API`;
+
+export const isResponseEmpty = (res: request.Response):boolean => {
```
Comment (Zizzamia) [inline src/resource/Resource.ts]: This should be `isInvalidResponse`

File: src/resource/Resource.ts [hunk 4]
```diff
@@ -3,7 +3,20 @@ import { Method } from '~/types';
...
+  const resIncludesJSON = res.type.includes('json');
+  // Empty is only valid when no response is expect (204)
+  const resNoResponse = res.text === '' && res.status === 204;
```
Comment (ntucker) [inline src/resource/Resource.ts]: suggestion: resEmptyIsExpected

**Comment by ntucker** [comment]
Thanks for the PR, I'd love to accept these improvements with the small feedback mentioned. :)

**Comment by Zizzamia** [comment]
@ntucker updated the PR 😄

**Pull request merged**